app/models/chargeback/consumption.rb

```diff
@@ -4,8 +4,15 @@ def initialize(start_time, end_time)
       @start_time, @end_time = start_time, end_time
     end
 
-    def hours_in_interval
-      @hours_in_interval ||= (@end_time - @start_time).round / 1.hour
+    def consumed_hours_in_interval
+      # Why we need this?
+      #   1) We cannot charge for hours until the resources existed (vm provisioned in the middle of month)
+      #   2) We cannot charge for future hours (i.e. weekly report on Monday, should charge just monday)
+      @consumed_hours_in_interval ||= begin
+                                        consuption_start = [@start_time, resource.try(:created_on)].compact.max
+                                        consumption_end = [Time.current, @end_time].min
```

Comment:
Should this use `Time.current.utc` to ensure the time is not in the local time zone?